.github/workflows/extract-translation-source-files.yml

```diff
@@ -254,6 +254,98 @@ jobs:
           # push changes to branch
           git push
 
+  # Non-Django, non-JS repositories with a generic extraction interface
+  generic-translations:
+    strategy:
+      # using max-parallel to avoid git push/pull issues when running in parallel
+      max-parallel: 1
+      matrix:
+        repo:
+          - tutor-contrib-aspects
+
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    needs: [setup-branch]
+
+    steps:
+      # Clones the openedx-translations repo
+      - name: clone openedx/openedx-translations
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.setup-branch.outputs.branch }}
+
+      # Clones the  repository
+      - name: clone openedx/${{ matrix.repo }}
+        uses: actions/checkout@v3
+        with:
+          repository: openedx/${{ matrix.repo }}
+          path: translations/${{ matrix.repo }}
+
+      # Sets up Python
+      - name: setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+
+      # Installs Python requirements from translations.txt
+      - name: install requirements
+        run: pip install -r requirements/translations.txt
+
+      # Extracts the translation source files
+      - name: extract translation source files
+        run: |
+          cd translations/${{ matrix.repo }}
+          make extract_translations
+
+      # Validate compilation of translation source files
+      - name: validate compilation of translation source files
+        run: |
+          cd translations/${{ matrix.repo }}
+          make compile_translations
+
+      # git adds only the translation source files, found in conf/locale/en
+      - name: git add the translation source files
+        id: add-sources
+        run: |
+          # set identity
+          git config --global user.email "translations-bot@openedx.org"
+          git config --global user.name "edx-transifex-bot"
+          # Change directory to translations/${{ matrix.repo }}
+          cd translations/${{ matrix.repo }}
+          # finds the directory containing the english locale usually located
+          # in */*/conf/locale/en
+          # This is slightly different from the python-translations line above
+          # in that it ignores top level dotfiles and searches further down the
+          # directory tree, which may be needed in these use cases.
+          EN_DIR=$(find ./* -type d -regex ".*conf\/locale\/en$")
```

Comment:
This raises a question for me. Would we rather follow the django pattern here or the js pattern? I think the js pattern is a little cleaner, but I'll leave the decision up to you.

### How we could implement "the js pattern"
* Have `make extract_translations` generate `transifex_input.yaml` (possibly in the root of the repo? no real preference on this as long as it's a path we're happy with using across all generic repos)
* `.gitignore` the `transifex_input.yaml` file in the repo that generates it
* For now (until we implement the full `atlas`/`openedx-translations` flow), have `make extract_translations` in `tutor-contrib-aspects` copy the `transifex_input.yaml` file to `tutoraspects/templates/aspects/apps/superset/conf/locale/en/locale.yaml`
* Instead of getting `EN_DIR` here, just grab the `transifex_input.yaml`

The main benefit I see with moving to this would be flexibility in where any given generic repo puts translation files. As long as `make extract_translations` generates `transifex_input.yaml` we're good to go

Reply:
Sure, that makes more sense. I'll work on that. Thanks!

Reply:
@brian-smith-tcril ok, both PRs are updated. Let me know what you think!